__tests__/user/user.mutation.test.js

```diff
@@ -0,0 +1,282 @@
+            'lastName', 'Tatsenyak'
+        );
+        expect(res.data.registerUser).toHaveProperty(
+            'email', 'tacjka34@gmail.com'
+        );
+        expect(res.data.registerUser).toHaveProperty('credentials');  
+        expect(res.data.registerUser).toHaveProperty('role');
+        expect(res.data.registerUser).toHaveProperty('registrationDate');
+    })
+
+    test('should throw error User with provided email already exist', async () => {
+        const res = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    registerUser(user: {
+                        firstName: "Petro" 
+                        lastName: "Tatsenyak"
+                        email: "tacjka34@gmail.com"
+                        password: "12345678Pt"
+                    }) {
+                        _id
+                        firstName
+                        lastName
+                        email
+                        role
+                        registrationDate
+                        credentials {
+                            tokenPass
+                        } 
+                    }
+                }
+            `
+        }).catch(err => err)
+
+        expect(res.graphQLErrors.length).toBe(1) 
+        expect(res.graphQLErrors[0].message).toBe('User with provided email already exists') 
+
+    })
+
+    test('should authorize and recive user token', async () => {
+        const res = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    loginUser(user: {
+                        email: "tacjka34@gmail.com"
+                        password: "12345678Pt"
+                    }) {
+                        _id
+                        firstName
+                        token
+                    }
+                }
+            `
+        })
+        
+        expect(res.data.loginUser).toHaveProperty('token');
+        
+        token = res.data.loginUser.token  
+    })
+
+    test('should throw error User with provided email not found', async () => {
+        const res = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    loginUser(user: {
+                        email: "udernotfound@gmail.com"
+                        password: "12345678Pt"
+                    }) {
+                        _id
+                        firstName
+                        token
+                    }
+                }
+            `
+        }).catch(err => err)
+    
+        expect(res.graphQLErrors.length).toBe(1) 
+        expect(res.graphQLErrors[0].message).toBe('User with provided email not found') 
+    })
+
+    test('should update user by id', async () => {
+        const res = await client.mutate({
+            mutation: gql`
+                mutation($userId: ID!) {       
+                    updateUserById(user: {
+                        firstName: "Updated",
```

Comment:
For the `user` input try to pass it not as a part of the query string, but as a separate variable. It will allow you to put it inside the plain js object and then use it in different places, especially in actual assertions to ensure that if you change user data in this object you don't need to change it manually in assertions too. Please, do it for all `users` tests. As an example, it is more for query tests but the common meaning is the same. You create user, get its fields, save it in some global variable and then when you get this user back you compare its returned fields in response with fields in your saved object

Reply:
Also, please, refactor all queries and mutations to use variables `$userID` etc, it is good practice

Reply:
fixed